### PR TITLE
javadoc fixups

### PR DESCRIPTION
**Pull request by rsgowman**
Addresses the comments in http://b/137376378.

I additionally attempted to find/fix some other similar issues while I was at it. In all cases, I used automated/semi-automated methods to find/fix rather than actually auditing the files, so there's probably many more things that could be addressed.

Generally, I've used `@code` for self-references (i.e. `{@code Blob}` within Blob.java) and `@link` otherwise.) But I haven't attempted to unify this across the codebase.

### CODE REVIEW

File: firebase-firestore/src/main/java/com/google/firebase/firestore/CollectionReference.java [hunk 1]
```diff
@@ -56,10 +56,10 @@ public String getId() {
   }
 
   /**
-   * Gets a DocumentReference to the document that contains this collection. Only subcollections are
-   * contained in a document. For root collections, returns null.
+   * Gets a {@link DocumentReference} to the document that contains this collection. Only
+   * subcollections are contained in a document. For root collections, returns null.
    *
-   * @return The DocumentReference that contains this collection or null if this is a root
+   * @return The {@link DocumentReference} that contains this collection or null if this is a root
```
Comment (wilhuff) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/CollectionReference.java]: This will add considerable visual noise for no actual benefit. I think we should mostly undo this, or in cases where we're concretely talking about the type and it's important mark those as `{@code}`. `{@link}` should be used to only to refer to types or methods that aren't otherwise implied by the types in the signature.

Look at the current [public method summary](https://firebase.google.com/docs/reference/android/com/google/firebase/firestore/CollectionReference#public-method-summary). Also look at the specific doc for getParent (a direct link to which doesn't seem to work). It's already contextually the case that any type mentioned in the signature is linked.

Here's how this [looks in the JDK](https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html#insert(int,%20java.lang.CharSequence,%20int,%20int)). In this example `CharSequence` is the type of a parameter of the method so it's merely marked as code, not linked.

By contrast, [this method](https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html#insert(int,%20boolean)) links to `String.valueOf(boolean)` because that link isn't implied by the types of the method.

In JDK javadoc, they frequently go even further and spell words out when referring to the concept behind the type rather than referring to the type name, e.g. [here they spell "character sequence"](https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html#append(java.lang.CharSequence) rather than `CharSequence`. This doesn't seem to be applied consistently though.

In any case, adding `{@link}` to all non-self types is overkill and distracting.

Adding `{@code}` to everything is also likely to be overkill though I don't mind that as much. Consider changing to just plain words in most cases when the specific type isn't important.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/CollectionReference.java [hunk 2]
```diff
@@ -56,10 +56,10 @@ public String getId() {
   }
 
   /**
-   * Gets a DocumentReference to the document that contains this collection. Only subcollections are
-   * contained in a document. For root collections, returns null.
+   * Gets a {@link DocumentReference} to the document that contains this collection. Only
+   * subcollections are contained in a document. For root collections, returns null.
```
Comment (wilhuff) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/CollectionReference.java]: When mentioning the `null` literal we should mark that with `{@code}`. See e.g.:

https://docs.oracle.com/javase/7/docs/api/java/io/File.html#getParentFile()

File: firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentChange.java [hunk 3]
```diff
@@ -110,26 +111,26 @@ public QueryDocumentSnapshot getDocument() {
   }
 
   /**
-   * The index of the changed document in the result set immediately prior to this DocumentChange
-   * (i.e. supposing that all prior DocumentChange objects have been applied). Returns -1 for
-   * 'added' events.
+   * The index of the changed document in the result set immediately prior to this {@code
+   * DocumentChange} (i.e. supposing that all prior {@code DocumentChange} objects have been
```
Comment (rachelsaunders) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentChange.java]: maybe "i.e. supposing" to "assuming"

File: firebase-firestore/src/main/java/com/google/firebase/firestore/FieldValue.java [hunk 4]
```diff
@@ -19,35 +19,38 @@
...
 @PublicApi
 public abstract class FieldValue {
 
   FieldValue() {}
 
   /**
-   * Returns the method name (e.g. "FieldValue.delete") that was used to create this FieldValue
-   * instance, for use in error messages, etc.
+   * Returns the method name (e.g. "FieldValue.delete") that was used to create this {@code
+   * FieldValue} instance, for use in error messages, etc.
```
Comment (rachelsaunders) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/FieldValue.java]: "e.g." to "for example,"

File: firebase-firestore/src/main/java/com/google/firebase/firestore/SnapshotMetadata.java [hunk 5]
```diff
@@ -35,10 +35,11 @@
   }
 
   /**
-   * @return true if the snapshot contains the result of local writes (e.g. set() or update() calls)
-   *     that have not yet been committed to the backend. If your listener has opted into metadata
-   *     updates (via `MetadataChanges.INCLUDE`) you will receive another snapshot with
-   *     `hasPendingWrites()` equal to false once the writes have been committed to the backend.
+   * @return true if the snapshot contains the result of local writes (e.g. {@code set()} or {@code
```
Comment (rachelsaunders) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/SnapshotMetadata.java]: "e.g." to "for example,"

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLitePersistence.java [hunk 6]
```diff
@@ -144,13 +144,13 @@ public void start() {
...
-              + " Keep in mind that multi-process Android apps execute the code in your"
+          "Failed to gain exclusive lock to the Cloud Firestore client's offline persistence. This"
+              + " generally means you are using Cloud Firestore from multiple processes in your"
+              + " app. Keep in mind that multi-process Android apps execute the code in your"
               + " Application class in all processes, so you may need to avoid initializing"
-              + " Firestore in your Application class. If you are intentionally using Firestore"
-              + " from multiple processes, you can only enable offline persistence (i.e. call"
-              + " setPersistenceEnabled(true)) in one of them.",
+              + " Cloud Firestore in your Application class. If you are intentionally using Cloud"
+              + " Firestore from multiple processes, you can only enable offline persistence (i.e. "
```
Comment (rachelsaunders) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLitePersistence.java]: "i.e." to "that is,"

File: firebase-firestore/src/main/java/com/google/firebase/firestore/remote/Datastore.java [hunk 7]
```diff
@@ -63,9 +63,9 @@
    * part of their app's dependencies.
    */
   static final String SSL_DEPENDENCY_ERROR_MESSAGE =
-      "The Firestore SDK failed to establish a secure connection. This is likely a problem with "
-          + "your app, rather than with Firestore itself. See https://bit.ly/2XFpdma for "
-          + "instructions on how to enable TLS on Android 4.x devices.";
+      "The Cloud Firestore SDK failed to establish a secure connection. This is likely a problem "
```
Comment (wilhuff) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/remote/Datastore.java]: This is wrong but the correct text is probably too wordy. This isn't a "Cloud Firestore SDK", this is the "Cloud Firestore for Firebase SDK" :-(. In the other error messages we refer to things like this "Cloud Firestore client's offline persistence". Maybe s/SDK/client/?

File: firebase-firestore/src/main/java/com/google/firebase/firestore/MetadataChanges.java [hunk 8]
```diff
@@ -17,7 +17,7 @@
 import com.google.firebase.annotations.PublicApi;
 
 /**
- * Indicates whether metadata-only changes (i.e. only {@code DocumentSnapshot.getMetadata()} or
+ * Indicates whether metadata-only changes (i.e.&nbsp;only {@code DocumentSnapshot.getMetadata()} or
```
Comment (wilhuff) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/MetadataChanges.java]: Was this intended?

File: firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentChange.java [hunk 9]
```diff
@@ -110,26 +111,26 @@ public QueryDocumentSnapshot getDocument() {
...
+   * DocumentChange} (assuming that all prior {@code DocumentChange} objects and the current {@code
+   * DocumentChange} object have been applied). Returns -1 for 'removed' events.
    */
   @PublicApi
   public int getNewIndex() {
     return newIndex;
   }
 
-  /** Creates the list of DocumentChanges from a ViewSnapshot. */
+  /** Creates the list of {@code DocumentChange}s from a {@code ViewSnapshot}. */
```
Comment (wilhuff) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentChange.java]: pluralizing with a "s" outside the `{@code block}` ends up rendering with mixed fonts in a single word and is ugly.

Consider rephrasing to "list of `{@code DocumentChange}` instances" (or "objects") or "list of document changes".

File: firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java [hunk 10]
```diff
@@ -419,17 +417,17 @@ public static void setLoggingEnabled(boolean loggingEnabled) {
   /**
    * Clears the persistent storage, including pending writes and cached documents.
    *
-   * <p>Must be called while the FirebaseFirestore instance is not started (after the app is
+   * <p>Must be called while the {@code FirebaseFirestore} instance is not started (after the app is
    * shutdown or when the app is first initialized). On startup, this method must be called before
-   * other methods (other than <code>setFirestoreSettings()</code>). If the FirebaseFirestore
+   * other methods (other than {@link #setFirestoreSettings()}). If the {@code FirebaseFirestore}
    * instance is still running, the <code>Task</code> will fail with an error code of <code>
```
[the remark this answers is not in the document]
Comment (rsgowman) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java]: done.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/MetadataChanges.java (hunk 8 above)
Comment (rsgowman) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/MetadataChanges.java]: Yes. Without this, javadoc cuts the summary fragment after the "i.e." (Javadoc assumes the first period followed by a space ends the summary fragment.) There aren't any good workarounds; adding nbsp tricks javadoc into not thinking there's a space there.

But Rachel's suggestions elsewhere provide another possibility; just spell it out in english. Done.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentChange.java (hunk 9 above)
Comment (rsgowman) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentChange.java]: Used 'document changes' (since I have to address this one individually.)
